cmd/test-bot.rb

```diff
@@ -1,132 +1,131 @@
-      switch "--only-formulae",
-             description: "Only run the formulae steps."
-      switch "--only-formulae-detect",
-             description: "Only run the formulae detection steps."
-      switch "--only-formulae-dependents",
-             description: "Only run the formulae dependents steps."
-      switch "--only-bottles-fetch",
-             description: "Only run the bottles fetch steps. This optional post-upload test checks that all " \
-                          "the bottles were uploaded correctly. It is not run unless requested and only needs " \
-                          "to be run on a single machine. The bottle commit to be tested must be on the tested " \
-                          "branch."
-      switch "--only-cleanup-after",
-             description: "Only run the post-cleanup step. Needs `--cleanup`."
-      comma_array "--testing-formulae=",
-                  description: "Use these testing formulae rather than running the formulae detection steps."
-      comma_array "--added-formulae=",
-                  description: "Use these added formulae rather than running the formulae detection steps."
-      comma_array "--deleted-formulae=",
-                  description: "Use these deleted formulae rather than running the formulae detection steps."
-      comma_array "--skipped-or-failed-formulae=",
-                  description: "Use these skipped or failed formulae from formulae steps for a " \
-                               "formulae dependents step."
-      conflicts "--only-formulae-detect", "--testing-formulae"
-      conflicts "--only-formulae-detect", "--added-formulae"
-      conflicts "--only-formulae-detect", "--deleted-formulae"
-      conflicts "--skip-dependents", "--only-formulae-dependents"
-      conflicts "--only-cleanup-before", "--only-setup", "--only-tap-syntax",
-                "--only-formulae", "--only-formulae-detect", "--only-formulae-dependents",
-                "--only-cleanup-after", "--skip-setup"
-    end
-  end
+      def run
+        setup_argv_and_env
 
-  def test_bot
-    setup_argv_and_env
+        # Keep this after the args parse to keep --help fast.
+        require_relative "../lib/test_bot"
 
-    args = test_bot_args.parse
+        Homebrew::TestBot.run!(args)
+      end
 
-    # Keep this after the .parse to keep --help fast.
-    require_relative "../lib/test_bot"
+      def setup_argv_and_env
+        return if ENV["GITHUB_ACTIONS"].blank? && ENV["GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED"].blank?
 
-    Homebrew::TestBot.run!(args)
-  end
-
-  def setup_argv_and_env
-    return if ENV["GITHUB_ACTIONS"].blank? && ENV["GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED"].blank?
-
-    ARGV << "--cleanup" << "--local"
-    ENV["HOMEBREW_COLOR"] = "1"
-    ENV["HOMEBREW_GITHUB_ACTIONS"] = "1"
+        ARGV << "--cleanup" << "--local"
+        ENV["HOMEBREW_COLOR"] = "1"
+        ENV["HOMEBREW_GITHUB_ACTIONS"] = "1"
```

Comment:
@dduugg This approach doesn't quite work. Any thoughts on how best to handle it?

Reply:
I see this went in, I'm not sure what your issue was. I know working with `ARGV` can be a pain, the `irb` command has some workarounds that might be helpful to reference.

Reply:
@dduugg Thanks for replying! Yeh, @Bo98 suggested just not doing this which was easier than expected.